serenata_toolbox/federal_senate/federal_senate_dataset.py

```diff
@@ -0,0 +1,83 @@
+import os.path
+from urllib.request import urlretrieve
+import numpy as np
+import pandas as pd
+
+class FederalSenateDataset:
+    URL = 'http://www.senado.gov.br/transparencia/LAI/verba/{}.csv'
+
+    def __init__(self, path):
+        self.path = path
+
+    def fetch(self):
+        urls = [self.URL.format(year) for year in range(2008, 2018)]
+        filename_from_url = lambda url: 'federal-senate-{}'.format(url.split('/')[-1])
```

Comment:
Having a `lambda` this way is not a best practice, but I can handle that. The real problem here is that using [`urllib.parse.urlsplit`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlsplit) and [`os.path.basename`](https://docs.python.org/3/library/os.path.html#os.path.basename) would be way safer ; )

Something among these lines:

```python
from urllib.urlparse import urlsplit

…

    def fetch(self):
        …
        url_paths = (urlsplit(url).year for url in urls)
        filenames = map(os.path.basename, url_paths)
        …
```



os.path.basename